spec/data-types-and-values.html

```diff
@@ -127,17 +127,20 @@ <h1>
         <emu-alg>
             1. Let _list_ be _argument_.[[Sequence]].
             1. Assert: _list_ is a List of ECMAscript language values.
-            1. Let _length_ be the length of _list_.
-            1. Let _ret_ be *"#["*.
-            1. Let _k_ be 0.
-            1. Repeat, while _k_ &lt; _len_,
-              1. Let _value_ be _list_[_k_].
+            1. Let _strings_ be a new empty List.
+            1. For each element _value_ of _list_, do
               1. If _value_ is a Symbol, Let _valueString_ be SymbolDescriptiveString(_value_).
               1. Else if _value_ is a String, Let _valueString_ by QuoteJSONString(_value_).
               1. Else, Let _valueString_ be ! ToString(_value_).
-              1. Set _ret_ to be the string-concatenation of _ret_ and _valueString_.
-              1. If _k_ &lt; _len_, Set _ret_ to be the string-concatenation of _ret_ and *", "*.
+              1. Append _valueString_ to the end of _strings_.
+            1. Let _len_ be the length of _strings_.
+            1. Let _ret_ be *"#["*.
+            1. Let _k_ be 0.
+            1. Repeat, while _k_ &lt; _len_,
+              1. Let _string_ be _strings_[_k_].
+              1. Set _ret_ to be the string-concatenation of _ret_ and _string_.
               1. Set _k_ to _k_ + 1.
+              1. If _k_ &lt; _len_, Set _ret_ to be the string-concatenation of _ret_ and *", "*.
```

Comment:
if this can't be done as single-line prose, maybe make an AO for `ListJoin`?

you'd be able to use it in the other algorithm also.